// @polygon.io/client-js/lib/rest/forex/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.forexClient = void 0;
const request_1 = require("../transport/request");
const lastQuoteForCurrencyPair_1 = require("./lastQuoteForCurrencyPair");
const historicForexTicks_1 = require("./historicForexTicks");
const aggregates_1 = require("./aggregates");
const snapshots_1 = require("./snapshots");
const realTimeCurrencyConversion_1 = require("./realTimeCurrencyConversion");
const forexClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    lastQuoteForCurrencyPair: request_1.auth(apiKey, lastQuoteForCurrencyPair_1.lastQuoteForCurrencyPair, apiBase),
    historicTicks: request_1.auth(apiKey, historicForexTicks_1.historicForexTicks, apiBase),
    realTimeCurrencyConversion: request_1.auth(apiKey, realTimeCurrencyConversion_1.realTimeCurrencyConversion, apiBase),
    // aggregates
    previousClose: request_1.auth(apiKey, aggregates_1.forexPreviousClose, apiBase),
    aggregates: request_1.auth(apiKey, aggregates_1.forexAggregates, apiBase),
    groupedDaily: request_1.auth(apiKey, aggregates_1.forexGroupedDaily, apiBase),
    // snapshots
    snapshotAllTickers: request_1.auth(apiKey, snapshots_1.forexSnapshotAllTickers, apiBase),
    snapshotGainersLosers: request_1.auth(apiKey, snapshots_1.forexSnapshotGainersLosers, apiBase)
});
exports.forexClient = forexClient;
exports.default = exports.forexClient;
