// @polygon.io/client-js/lib/rest/crypto/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cryptoClient = void 0;
const request_1 = require("../transport/request");
const dailyOpenClose_1 = require("./dailyOpenClose");
const cryptoExchanges_1 = require("./cryptoExchanges");
const lastTradeForACryptoPair_1 = require("./lastTradeForACryptoPair");
const snapshots_1 = require("./snapshots");
const aggregates_1 = require("./aggregates");
const historicCryptoTrades_1 = require("./historicCryptoTrades");
const cryptoClient = (apiKey, apiBase = "https://api.polygon.io") => ({
    dailyOpenClose: request_1.auth(apiKey, dailyOpenClose_1.cryptoDailyOpenClose, apiBase),
    exchanges: request_1.auth(apiKey, cryptoExchanges_1.cryptoExchanges, apiBase),
    lastTradeForPair: request_1.auth(apiKey, lastTradeForACryptoPair_1.lastTradeForCryptoPair, apiBase),
    historicTrades: request_1.auth(apiKey, historicCryptoTrades_1.historicCryptoTrades, apiBase),
    // snapshots
    snapshotSingleTicker: request_1.auth(apiKey, snapshots_1.cryptoSnapshotSingleTicker, apiBase),
    snapshotAllTickers: request_1.auth(apiKey, snapshots_1.cryptoSnapshotAllTickers, apiBase),
    snapshotGainersLosers: request_1.auth(apiKey, snapshots_1.cryptoSnapshotGainersLosers, apiBase),
    snapshotSingleTickerFullBook: request_1.auth(apiKey, snapshots_1.cryptoSnapshotSingleTickerFullBook, apiBase),
    // aggregates
    previousClose: request_1.auth(apiKey, aggregates_1.cryptoPreviousClose, apiBase),
    aggregates: request_1.auth(apiKey, aggregates_1.cryptoAggregates, apiBase),
    groupedDaily: request_1.auth(apiKey, aggregates_1.cryptoGroupedDaily, apiBase)
});
exports.cryptoClient = cryptoClient;
exports.default = exports.cryptoClient;
